portal/src/graphql/adminapi/DeleteGroupDialog.tsx

```diff
@@ -0,0 +1,115 @@
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import {
+  Dialog,
+  DialogFooter,
+  IDialogContentProps,
+  IModalProps,
+} from "@fluentui/react";
+import { useSystemConfig } from "../../context/SystemConfigContext";
+import { Context, FormattedMessage } from "@oursky/react-messageformat";
+import PrimaryButton from "../../PrimaryButton";
+import DefaultButton from "../../DefaultButton";
+import ErrorDialog from "../../error/ErrorDialog";
+import { useDeleteGroupMutation } from "./mutations/deleteGroupMutation";
+
+export interface DeleteGroupDialogData {
+  groupID: string;
+  groupName: string | null;
+  groupKey: string;
+}
+
+interface DeleteGroupDialogProps {
+  data: DeleteGroupDialogData | null;
+  onDismiss: (isDeleted: boolean) => void;
+  onDismissed?: () => void;
+}
+
+function useSnapshotData(data: DeleteGroupDialogData | null) {
```

Comment:
:sadfrog: instead of copying this hook again, can we create a new file to implement this hook in a generic way and reuse it?